src/hooks/useClerkAttachAuthInterceptorsToAxios.tsx

```diff
@@ -0,0 +1,91 @@
+import { useAuth, useOrganization } from "@clerk/nextjs";
+import {
+  Auth,
+  CanaryChecker,
+  Config,
+  ConfigDB,
+  IncidentCommander,
+  Logs,
+  Rback,
+  redirectToLoginPageOnSessionExpiry,
+  Snapshot
+} from "@flanksource-ui/api/axios";
+import { toastError } from "@flanksource-ui/components/Toast/toast";
+import { useEffect } from "react";
+
+const allAxiosInstances = [
+  Auth,
+  IncidentCommander,
+  Logs,
+  CanaryChecker,
+  Config,
+  ConfigDB,
+  Rback,
+  Snapshot
+];
+
+export default function useClerkAttachAuthInterceptorsToAxios() {
+  const { getToken } = useAuth();
+  const { organization } = useOrganization();
+
+  // we should only get to this point if the user is logged in and has an
+  // organization set
+  const backendUrl = organization?.publicMetadata.backend_url as string;
+
+  useEffect(() => {
+    const interceptorsRequestCleanups: number[] = [];
+    const interceptorsResponseCleanups: number[] = [];
+
+    allAxiosInstances.forEach((axiosInstance) => {
+      if (!backendUrl) {
+        return;
+      }
+
+      // register axios interceptors here
+      const requestInterceptor = axiosInstance.interceptors.request.use(
+        async (config) => {
```

Comment:
Have we confirmed that this only happens once, and not on every backend request? What happens when the token expires or is it a long lived token ?

Reply:
I created a `Backend` JWT template with a 30 day expiry that we should use, but we still shouldn't be requesting it every time.

Reply:
@moshloop the getAuth method should cache the token within the token TTL, you would need to skipCache explicitly. In terms of renewal, clerk should handle that automatically for us, and we will always get an update token when we request, either from cache or a fresh if the TTL is expired.